planemo/galaxy/activity.py

```diff
@@ -456,7 +446,15 @@ def _history_content_download(self, history_id, dataset_id, to_path, filename=No
             data["filename"] = filename
 
         r = user_gi.make_get_request(url, params=data, stream=True, timeout=user_gi.timeout)
-        r.raise_for_status()
+        try:
+            r.raise_for_status()
+        except HTTPError as e:
+            # When a job fails abruptly the object store may not contain a dataset,
+            # and that results in an internal server error on the Galaxy side.
+            # We don't want this to break the rest of the test report.
+            # Should probably find a way to propagate this back into the report.
+            ctx.vlog(f"Failed to download history content at URL {url}, exception: {e}")
+            return
```

Comment:
The job needs to fail right? You can run test assertions on this.

Reply:
If we're reaching this point the job has already failed (most likely), and the test assertions would run on an empty file, so they should still produce the failure. Checkout line 380 and 363, where we're also skipping outputs.